gomint-api/src/main/java/io/gomint/math/BlockPosition.java

```diff
@@ -54,17 +52,11 @@ public Vector toVector() {
     }
 
     public BlockPosition add( BlockPosition other ) {
-        this.x += other.x;
-        this.y += other.y;
-        this.z += other.z;
-        return this;
+        return new BlockPosition(this.x + other.x, this.y + other.y, this.z + other.z);
```

Comment:
This type of changes doesn't break the api, because there is no api doc, but plugins might have relied upon it modifying the existing object.
I suggest having some sort of warning somewhere that these methods have changed their behaviour.